lib/rollbar/plugin.rb

```diff
@@ -15,31 +17,73 @@ def initialize(name)
       @name = name
       @dependencies = []
       @callables = []
+      @revert_callables = []
       @loaded = false
+      @on_demand = false
+    end
+
+    def load_on_demand
+      @on_demand = true
     end
 
     def configuration
       Rollbar.configuration
     end
 
+    def load_scoped!(transparent = false)
```

Comment:
To me, the `transparent` flag seems unnecessary, and the `transparent == false` case causes plugin dependency statements to have unintended side effects. (By causing code outside the plugin to not run.) The current use case relies on the `transparent == true` case, which makes sense. Do you have a use case in mind for the other behavior?

Reply:
The `transparent == false` would be used if you want the provided logic to only be run if the plugin is successfully loaded. Which I assume would be in the majority of situations, other than this ;) Does it make sense?

Reply:
Yes, thank you.